arkane/common.py

```diff
@@ -31,12 +31,233 @@
+                self.symmetry_number = species.symmetryNumber
+            if species.transportData is not None:
+                self.transport_data = species.transportData  # called `collisionModel` in Arkane
+            if species.energyTransferModel is not None:
+                self.energy_transfer_model = species.energyTransferModel
+            if species.thermo is not None:
+                self.thermo = species.thermo.as_dict()
+                thermo_data = species.getThermoData()
+                h298 = thermo_data.getEnthalpy(298) / 4184.
+                s298 = thermo_data.getEntropy(298) / 4.184
+                cp = dict()
+                for t in [300,400,500,600,800,1000,1500,2000,2400]:
+                    temp_str = '{0} K'.format(t)
+                    cp[temp_str] = '{0:.2f}'.format(thermo_data.getHeatCapacity(t) / 4.184)
+                self.thermo_data = {'H298': '{0:.2f} kcal/mol'.format(h298),
+                                    'S298': '{0:.2f} cal/mol*K'.format(s298),
+                                    'Cp (cal/mol*K)': cp}
+
+    def update_xyz_string(self):
+        if self.conformer is not None and self.conformer.number is not None:
+            # generate the xyz-format string from the Conformer coordinates
+            xyz_string = '{0}\n{1}'.format(len(self.conformer.number.value_si), self.label)
+            for i, coorlist in enumerate(self.conformer.coordinates.value_si):
+                for element in elementList:
+                    if element.number == int(self.conformer.number.value_si[i]):
+                        element_symbol = element.symbol
+                        break
+                else:
+                    raise ValueError('Could not find element symbol corresponding to atom number {0}'.format(
+                        self.conformer.number.value_si[i]))
+                xyz_string += '\n{0} {1} {2} {3}'.format(element_symbol,
+                                                         coorlist[0],
+                                                         coorlist[1],
+                                                         coorlist[2])
+        else:
+            xyz_string = ''
+        return xyz_string
+
+    def save_yaml(self, path):
+        """
+        Save the species with all statMech data to a .yml file
+        """
+        if not os.path.exists(os.path.join(os.path.abspath(path),'ArkaneSpecies', '')):
+            os.mkdir(os.path.join(os.path.abspath(path),'ArkaneSpecies', ''))
+        valid_chars = "-_.()<=>+ %s%s" % (string.ascii_letters, string.digits)
+        filename = os.path.join('ArkaneSpecies',
+                                ''.join(c for c in self.label if c in valid_chars) + '.yml')
+        full_path = os.path.join(path, filename)
+        with open(full_path, 'w') as f:
+            yaml.dump(data=self.as_dict(), stream=f, canonical=False)
+        # remove empty lines from the file (multi-line strings have excess new line brakes for some reason):
```

Comment:
Thanks for the link above to the solution on stackoverflow re extra line breaks. The solution there uses a string representer, but we no longer use those since we don't `evaluate` any more. I'll implement your suggestion here.